### PR TITLE
Add from to use_helpers to add macro like syntax

### PR DESCRIPTION
**Pull request by reeganviljoen**
### What are you trying to accomplish?

`use_helepers` is powerful but as @joelhawksley mentioned in #1987

> I think we might be better off having this be a macro folks can set on a component, likely on their ApplicationComponent.

So this is my attempt at adding a macro syntax by allowing helpers to be defined from specific helper modules

### What approach did you choose and why?

I had two make use of two private methods one for the usage of defining a method from `__vc_original_view_context` or one from a specified module witch requires some more complex meta programming to change the ownership of the method to the component.

### CODE REVIEW

**Comment by reeganviljoen** [comment]
@joelhawksley @camertron @Spone it seems Rails main is broken again, I will investigate a fix tomorrow evening

**Comment by reeganviljoen** [comment]
@joelhawksley @boardfish @camertron I know I haven't filled an issue and i am mostly looking for any feedback on this, I feel that this makes this api largely more complete and  fits its purpose of making the usage of helpers more explicit. 🙏

File: docs/guide/helpers.md [hunk 1]
```diff
@@ -69,6 +69,18 @@ class UseHelpersComponent < ViewComponent::Base
 end
 ```
 
+Methods defined in helper modules that are not available in the component can be included individually using the `from:` keyword:
+
+```ruby
+class UserComponent < ViewComponent::Base
+  use_helpers :icon, from: IconHelper
```
Comment (Spone) [inline docs/guide/helpers.md:76]: I guess it would make sense to support the singular form `use_helper` as well?

Comment (reeganviljoen) [reply docs/guide/helpers.md]: @Spone I have added this recommendation, as it seems like a great fit

**Comment by joelhawksley** [comment]
@reeganviljoen - @camertron and I think this generally looks good, mind getting CI green?

**Comment by reeganviljoen** [comment]
@joelhawksley I will find time tomorrow to fix it

**Comment by reeganviljoen** [comment]
@joelhawksley I was able to get CI green by reverting #2039 , in the interesting of not opening another pr I included it here

File: docs/guide/helpers.md [hunk 2]
```diff
@@ -59,16 +59,40 @@ By default, ViewComponents don't have access to helper methods defined externall
...
   erb_template <<-ERB
     <div class="icon">
-      <%= icon :user %>
+      <%= icon? ? icon(:user) : icon(:guest) %>
     </div>
   ERB
 end
 ```
 
+Methods defined in helper modules that aren't available in the component can be included individually using the `from:` keyword:
```
Comment (joelhawksley) [inline docs/guide/helpers.md:72]:
```suggestion
Use the `from:` keyword to include individual methods defined in helper modules not available in the component:
```

File: docs/guide/helpers.md [hunk 3]
```diff
@@ -59,16 +59,40 @@ By default, ViewComponents don't have access to helper methods defined externall
...
+class UserComponent < ViewComponent::Base
+  use_helpers :icon, :icon?, from: IconHelper
+
+  def profile_icon
+    icon? ? icon(:user) : icon(:guest)
+  end
+end
+```
+
+The singular version `use_helper` is also available for a single helper:
```
Comment (joelhawksley) [inline docs/guide/helpers.md:84]:
```suggestion
The singular version `use_helper` is also available:
```

**Review by joelhawksley — APPROVED** [review]

**Pull request merged**